rm-portal/src/test/java/org/ow2/proactive_grid_cloud_portal/rm/client/monitoring/views/compact/CompactFlowPanelOwnTest.java

```diff
@@ -0,0 +1,146 @@
+            for (Tile nodeTile : nodes(nodeSourceTile.getNodesource().getSourceName(),
+                                       hostTile.getHost().getHostName(),
+                                       2)) {
+                compactFlowPanelOwn.drawNode(nodeTile, hostTile);
+            }
+        }
+        assertEquals(4, compactFlowPanelOwn.getTilesNumber());
+        compactFlowPanelOwn.remove(compactFlowPanelOwn.getModel().get(0).getHosts().get(0).getNodes().get(0));
+        assertEquals(3, compactFlowPanelOwn.getTilesNumber());
+        compactFlowPanelOwn.remove(compactFlowPanelOwn.getModel().get(0).getHosts().get(0).getNodes().get(0));
+        assertEquals(0, compactFlowPanelOwn.getTilesNumber());
+        verify(compactFlowPanelOwn, times(4)).remove(anyInt());
+    }
+
+    @Test
+    public void testNodeRemoval() {
+        List<NodeSource.Host.Node> nodes = new ArrayList<>();
+        for (Tile nodeSourceTile : nodeSources(5)) {
+            compactFlowPanelOwn.drawNodeSource(nodeSourceTile);
+
+            for (Tile hostTile : hosts(nodeSourceTile.getNodesource().getSourceName(), 5)) {
+                for (Tile nodeTile : nodes(nodeSourceTile.getNodesource().getSourceName(),
+                                           hostTile.getHost().getHostName(),
+                                           5)) {
+                    nodes.add(nodeTile.getNode());
+                    compactFlowPanelOwn.drawNode(nodeTile, hostTile);
+                }
+            }
+        }
+
+        // remove all nodes,
+        // dangling hosts also should be removed automatically
+        // dangling nodesources also should be removed automatically
+        for (NodeSource.Host.Node node : nodes) {
+            compactFlowPanelOwn.remove(node);
+        }
+
+        long totalCount = 5 * (1 + 5 * (1 + 5));
+        verify(compactFlowPanelOwn, times((int) totalCount)).remove(anyInt());
+
+        assertEquals(0, compactFlowPanelOwn.getTilesNumber());
+
+    }
+
+    @Test
+    public void testAddANode() {
+        final Tile nodeTile = node("SOURCEname", "local", "httpptth");
+        final Tile hostTile = host(nodeTile.getNode().getSourceName(), nodeTile.getNode().getHostName());
+        final Tile nodeSourceTile = nodeSource(nodeTile.getNode().getSourceName());
+        compactFlowPanelOwn.drawNodeSource(nodeSourceTile);
+
+        verify(compactFlowPanelOwn, times(0)).insert(any(Widget.class), anyInt());
+        assertEquals(0, compactFlowPanelOwn.getTilesNumber());
+
+        compactFlowPanelOwn.drawNode(nodeTile, hostTile);
+
+        verify(compactFlowPanelOwn, times(1)).insert(eq(nodeSourceTile), eq(0));
+        verify(compactFlowPanelOwn, times(1)).insert(eq(hostTile), eq(1));
+        verify(compactFlowPanelOwn, times(1)).insert(eq(nodeTile), eq(2));
+
+        assertEquals(3, compactFlowPanelOwn.getTilesNumber());
+    }
+
+    @Test(expected = NullPointerException.class)
+    public void testAddANode1() {
```

Comment:
It would be better to have a descriptive name here. It would be easier to identify the issue if the test fails. Moreover the method name is not used so it can be long.

Reply:
Done